### PR TITLE
Update gradle

### PR DESCRIPTION
**Pull request by fourlastor**
Note: `androidpublish.gradle` seemed unused so I just removed it (I can update that too if necessary instead)

### CODE REVIEW

**Comment by fourlastor** [comment]
Hmm. I need to update the workflow files as well

File: .github/workflows/compile.yml [hunk 1]
```diff
@@ -16,11 +16,11 @@ jobs:
...
+    - name: Set up JDK 11
       uses: actions/setup-java@v1
       with:
-        java-version: 1.8
+        java-version: 11
     - name: Grant execute permission for gradlew
       run: chmod +x gradlew
     - name: Local install
-      run: ./gradlew clean uploadArchives -PSNAPSHOT=true -PLOCAL=true
+      run: ./gradlew clean publish -PSNAPSHOT=true -PLOCAL=true
```
Comment (MrStahlfelge) [inline .github/workflows/compile.yml:26]: Should be changed in readme file as well

File: core-mapping/build.gradle [hunk 2]
```diff
@@ -1,27 +1,25 @@
-apply plugin : 'java-library'
+plugins {
+	id 'com.github.johnrengelman.shadow' version '8.1.1'
```
Comment (MrStahlfelge) [inline core-mapping/build.gradle]: Do we really need it? It is usually used to produce a fat jar

File: publish.gradle [hunk 3]
```diff
@@ -37,106 +37,84 @@ def getRepositoryPassword() {
...
-    }
-
-    artifacts {
-        archives libraryJar
-        archives sourcesJar
-        archives javadocJar
-    }
 
+signing {
+    required { isReleaseBuild() && gradle.taskGraph.hasTask("uploadArchives") }
```
Comment (MrStahlfelge) [inline publish.gradle]: publish

File: publish.gradle [hunk 4]
```diff
@@ -37,106 +37,84 @@ def getRepositoryPassword() {
...
+    sign publishing.publications.maven
```
Comment (MrStahlfelge) [inline publish.gradle:120]: Something does not work yet with signing, it is producing a NullPointerException on build

**Review by MrStahlfelge — CHANGES_REQUESTED** [review]
First of all, thanks for this contribution.
Something does not work with signing yet, can you take a look at it? You don't need signing files, just try to build without -PSNAPSHOT

File: core-mapping/build.gradle (hunk 2 above)
Comment (fourlastor) [reply core-mapping/build.gradle]: I don't think we do. I noticed there was a custom config to achieve the same, but I think we can simply remove it

File: publish.gradle (hunk 4 above)
Comment (fourlastor) [reply publish.gradle:120]: What did you run to test it? I ran `gradlew clean publish -PLOCAL=true` now and it seems fine

File: publish.gradle (hunk 3 above)
Comment (fourlastor) [reply publish.gradle]: [128879e](https://github.com/MrStahlfelge/gdx-controllerutils/pull/29/commits/128879ef5e71357d70f18ee698b7221ed2134358)

File: core-mapping/build.gradle (hunk 2 above)
Comment (fourlastor) [reply core-mapping/build.gradle]: [6649061](https://github.com/MrStahlfelge/gdx-controllerutils/pull/29/commits/664906111caaae23d415b7730db2469468ab5c43)

**Comment by MrStahlfelge** [comment]
Just try to publish to Maven:
`./gradlew clean publish -PNEXUS_USERNAME=Something -PNEXUS_PASSWORD=Something`

It's been some time since I published and I might have forgotten something

**Comment by fourlastor** [comment]
This gives me an error, I think it's missing `-PRELEASE` to work (which works fine for me, excluding http issues)

This is because the following block is behind an `if`:

```gradle
                // This won't be set without parameters
                if (isReleaseBuild()) {
                    url = getReleaseRepositoryUrl()
                } else if (isSnapshotBuild()) {
                    url = getSnapshotRepositoryUrl()
                }
```

**Comment by MrStahlfelge** [comment]
Thanks, that's what I forgot

**Review by MrStahlfelge — APPROVED** [review]

**Pull request merged**